### ISSUE TITLE
FR: render unrendered fragments on demand only

### ISSUE THREAD

**Issue by benasocj** [koreader/koreader#10392]
**Does your feature request involve difficulty completing a task? Please describe.**
First of all, the new "partial rerenderings" feature is great - thanks for that! At the same time it opens doors for further enhancements :blush::

I have a very big book with 2000 pages. Sometimes I want to increase the font size only for my current reading session and change it in the next session again (depending if I'm sitting or walking during my reading session). For such a case I don't need the entire book to be rerendered in the background (which consumes lots of battery power). Instead, it would suffice to rerender the current fragment only as I'm likely to change the font size in my next session again anyway.

**Describe the solution you'd like**
It would be great if the automatic rerendering of the remaining fragments in the background could be turned off, perhaps even on a per-book basis. That means that once I read into the next unrendered fragment, it would only then be rerendered on demand (for a smoother user experience this solution could be extended to rerender the previous and next fragments in the background so that the user doesn't have to wait for the rendering once he reads into an unrendered fragment).

**Comment by poire-z** [comment]
> First of all, the new "partial rerenderings" feature is great - thanks for that!

Thank you :)

> At the same time it opens doors for further enhancements 😊:

No it doesn't ! :)

See https://github.com/koreader/koreader/issues/9087#issuecomment-1407750355 (and around).

TLDR: when partially rendered, lots of things work, but other important things are disabled (statistics) and broken (book map, footer info ...)
Making that partial state persistent and usable would be either impossible or very complicated - or just having some feature not available. But anyway, this would need lots of change to KOReader frontend stuff to make it clear some stuff are bad or not available. The cost of that doesn't amount for the (yours:) niche benefit.
And the fact that there is this simple solution of background full rerendering and reloading the document from cache makes it so obvious it's the way to go, that we don't have to think much more about your FR ! Sorry :)

If you really want to have what you wish - for yourself, and live with the side effects, I'm quite sure you could user-patch some frontend/apps/reader/modules/readerrolling.lua methods to not trigger that full rerendering and not show the icons - and live with that unsane state until you trigger a document reloading via a gesture - which should do the proper rerendering when the document is reloaded.

**Comment by benasocj** [comment]
> makes it so obvious it's the way to go, that we don't have to think much more about your FR ! Sorry :)

Sure, that makes sense and I totally understand! Thank you very much for your detailed answer and for all your hard work you constantly seem to invest into making KOReader for what it is :blush:.
